test/unit/canvas.js

```diff
@@ -1925,38 +1856,14 @@
     var circle = new fabric.Circle({ radius: 50, left: 50, top: 50 });
     canvas.add(rect, circle);
     var svg = canvas.toSVG();
-
-    canvas.setActiveObject(new fabric.ActiveSelection([rect, circle]));
+    const activeSelection = canvas.getActiveSelection();
+    activeSelection.add(rect, circle);
+    canvas.setActiveObject(activeSelection);
     var svgWithActiveGroup = canvas.toSVG();
 
     assert.equal(svg, svgWithActiveGroup);
   });
 
-  QUnit.test('active group objects reordering', function(assert) {
-    var rect1 = new fabric.Rect({ width: 30, height: 30, left: 130, top: 130 });
-    var rect2 = new fabric.Rect({ width: 50, height: 50, left: 100, top: 100 });
-    var circle1 = new fabric.Circle({ radius: 10, left: 60, top: 60 });
-    var circle2 = new fabric.Circle({ radius: 50, left: 50, top: 50 });
-    canvas.add(rect1, rect2, circle1, circle2);
-    assert.equal(canvas._objects[0], rect1);
-    assert.equal(canvas._objects[1], rect2);
-    assert.equal(canvas._objects[2], circle1);
-    assert.equal(canvas._objects[3], circle2);
-    var aGroup = new fabric.ActiveSelection([rect2, circle2, rect1, circle1], { canvas: canvas });
-    // before rendering objects are ordered in insert order
-    assert.equal(aGroup._objects[0], rect2);
-    assert.equal(aGroup._objects[1], circle2);
-    assert.equal(aGroup._objects[2], rect1);
-    assert.equal(aGroup._objects[3], circle1);
-    canvas.setActiveObject(aGroup)
-    canvas.renderAll();
-    // after rendering objects are ordered in canvas stack order
-    assert.equal(aGroup._objects[0], rect1);
-    assert.equal(aGroup._objects[1], rect2);
-    assert.equal(aGroup._objects[2], circle1);
-    assert.equal(aGroup._objects[3], circle2);
-  });
-
```

Comment:
@asturur please explain this to me as part of the discussion of click order vs. stack order
I believe this is what convinced me it was a bug, that rendering changed the order of the objects in active selection.
Now with this PR active selection will respect both click order and stack order

Reply:
I would say before it didn't respect either